txpool/grpc_server.go

```diff
@@ -68,35 +76,76 @@ func NewGrpcServer(ctx context.Context, txPool txPool, db kv.RoDB, chainID uint2
 func (s *GrpcServer) Version(context.Context, *emptypb.Empty) (*types2.VersionReply, error) {
 	return TxPoolAPIVersion, nil
 }
-func convertSubPoolType(t SubPoolType) txpool_proto.AllReply_Type {
+func convertSubPoolType(t SubPoolType) txpool_proto.Tx_Type {
 	switch t {
 	case PendingSubPool:
-		return txpool_proto.AllReply_PENDING
+		return txpool_proto.Tx_PENDING
 	case BaseFeeSubPool:
-		return txpool_proto.AllReply_PENDING
+		return txpool_proto.Tx_BASE_FEE
 	case QueuedSubPool:
-		return txpool_proto.AllReply_QUEUED
+		return txpool_proto.Tx_QUEUED
 	default:
 		panic("unknown")
 	}
 }
+
 func (s *GrpcServer) All(ctx context.Context, _ *txpool_proto.AllRequest) (*txpool_proto.AllReply, error) {
 	tx, err := s.db.BeginRo(ctx)
 	if err != nil {
 		return nil, err
 	}
 	defer tx.Rollback()
 	reply := &txpool_proto.AllReply{}
-	reply.Txs = make([]*txpool_proto.AllReply_Tx, 0, 32)
+	reply.Txs = make([]*txpool_proto.Tx, 0, 32)
 	if err := s.txPool.deprecatedForEach(ctx, func(rlp, sender []byte, t SubPoolType) {
-		reply.Txs = append(reply.Txs, &txpool_proto.AllReply_Tx{
+		reply.Txs = append(reply.Txs, &txpool_proto.Tx{
 			Sender: sender,
 			Type:   convertSubPoolType(t),
 			RlpTx:  rlp,
 		})
-	}, tx); err != nil {
+	}, nil, tx); err != nil {
+		return nil, err
+	}
+	return reply, nil
+}
+
+func (s *GrpcServer) Search(ctx context.Context, req *txpool_proto.SearchRequest) (*txpool_proto.SearchReply, error) {
+	tx, err := s.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	reply := &txpool_proto.SearchReply{
+		Txs: make([]*txpool_proto.Tx, 0, 32),
+	}
+
+	err = s.txPool.deprecatedForEach(
```

Comment:
There is method `pool.all.ascend` - example of how to iterate over txs of given sender (without iteration over all txs). Please use it or do similar. (can make "sender filter" required - if you like).